cms/djangoapps/contentstore/views/course.py

```diff
@@ -283,6 +293,32 @@ def course_rerun_handler(request, course_key_string):
             })
 
 
+@login_required
+@ensure_csrf_cookie
+@require_GET
+def course_index_handler(request, course_key_string):
+    """
+    The restful handler for course indexing.
+    GET
+        html: return status of indexing task
+    """
+    # Only global staff (PMs) are able to rerun courses during the soft launch
+    if not GlobalStaff().has_user(request.user):
+        raise PermissionDenied()
+    course_key = CourseKey.from_string(course_key_string)
+    with modulestore().bulk_operations(course_key):
+        error = reindex_course_and_check_access(course_key, request.user)
```

Comment:
In this case error is just a list of stuff that could not be indexed. They are returned to a client, but I don't believe they should throw exception and block indexing process. These can happen on multiple stuff, broken video, broken subtitles, deleted data.. Seemed to me it would be enough to just list them on UI?